src/secret.rs

```diff
@@ -35,7 +36,7 @@ macro_rules! impl_display_secret {
 
                 hasher.write(DEBUG_HASH_TAG);
```

Comment:
I believe the test fails because `DefaultHasher::new()` above returns hasher seeded with a different random seed each time. Which also means this implementation is actually wrong if we want to allow comparing different secrets.

Reply:
Interesting, thanks for debugging it for me. So `shared_secret_point` is non-trivial to use then. I'll put some more work into it tomorrow. Cheers.

Reply:
On further thought, I don't get that, the same data hashed multiple times should give the same output irrespective of the hashing implementation. I haven't looked at the code yet, I was just pondering your message.

Reply:
`std` hasher is one intended for `HashMap` and implements DoS protection by randomizing among other things. It produces same hash for same input for same instances of the hasher but since we call `new` on each debug we're randomizing it.

> if we want to allow comparing different secrets

I meant humans comparing, not computers here. Only the `Debug` implementation is wrong.
Unless we want to implement our own hasher or store `DefaultHasher` in static variable I suggest to just remove the hash when compiled without `bitcoin_hashes`.